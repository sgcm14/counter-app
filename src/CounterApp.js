import { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 10 }) => {

  const [counter, setCounter] = useState(value); // []

  const handleAdd = (e) =>{
    // console.log(e)
    setCounter (counter +1);
    //setCounter( (c) => c+1);
  }

  const handleMinus = (e) =>{
    // console.log(e)
    setCounter (counter -1);
    //setCounter( (c) => c-1);
  }

  const handleReset = (e) =>{
    // console.log(e)
    setCounter (value);
    //setCounter( (c) => c-1);
  }

  return (
    <>
      <h1>CounterApp</h1>
      <p>{counter}</p>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleMinus}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
