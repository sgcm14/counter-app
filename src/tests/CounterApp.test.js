import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en <CounterApp />", () => {
  let wrapper = shallow(<CounterApp />);
  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("debe de mostrar <CounterApp /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el valor por defecto de 100", () => {
    const numero = 100;
    const wrapper = shallow(<CounterApp value={numero} />);

    const textoParrafo = wrapper.find("p").text();

    expect(parseInt(textoParrafo)).toEqual(numero);
  });

  test("debe de incrementar con el botón +1", () => {
    wrapper.find("button").at(0).simulate("click");
    const counterText = wrapper.find("p").text();

    expect(parseInt(counterText)).toEqual(11);
  });

  test("debe de disminuir con el botón -1", () => {
    wrapper.find("button").at(2).simulate("click");
    const counterText = wrapper.find("p").text();

    expect(parseInt(counterText)).toEqual(9);
  });

  test("debe de colocar el valor por defecto con el btn reset", () => {
    const numero = 105;
    const wrapper = shallow(<CounterApp value={numero} />);

    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");

    const counterText = wrapper.find("p").text();

    expect(parseInt(counterText)).toEqual(numero);
  });
});
