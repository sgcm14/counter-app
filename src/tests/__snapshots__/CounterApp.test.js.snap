// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
